app/helpers/application_helper.rb

```diff
@@ -316,20 +316,33 @@ def link_to_chart(name, values)
     link_to icon_tag('signal'), url, target: :blank
   end
 
-  # show which stages this reference is deploy(ed+ing) to
+  # Show which stages this reference has been or is currently being deployed to.
   def deployed_or_running_list(stages, reference)
-    html = "".html_safe
-    stages.each do |stage|
-      next unless deploy = stage.deployed_or_running_deploy
-      next unless deploy.references?(reference)
-      label = (deploy.active? ? "label-warning" : "label-success")
-
-      text = "".html_safe
-      text << stage.name
-      html << content_tag(:span, text, class: "label #{label} release-stage")
-      html << " "
+    pieces = stages.map do |stage|
+      # The first deploy is the most recent one.
+      deploy = stage.deploys.where(reference: reference).first
+
+      label = if deploy.nil?
+        nil
+      elsif deploy.running?
+        "label-warning"
+      elsif deploy.succeeded?
+        if deploy == stage.last_deploy
+          "label-success"
+        else
+          # Deploy is neither active nor is it the last successful one, but it
+          # succeeded in the past.
+          "label-default"
+        end
+      end
+
+      if label.present?
```

Comment:
prefer consistent `if thing` instead of `if thing.nil?` + `if thing.present?`

Reply:
:+1: